backend/src/firemerge/statement/parser.py

```
@@ -96,13 +96,16 @@ def _parse_rows(
         if join_col := self.role_cols.get(ColumnRole.DOC_NUMBER):
             assert iban_col is not None
             rows = list(rows)
-            join_rows = {
-                row[join_col.index]: row
-                for row in rows
-                if row[join_col.index] and row[iban_col.index] != self.account.iban
-            }
+            join_rows = {}
+            duplicates = []
+            for row in rows:
+                if (doc_number := row[join_col.index]) and row[iban_col.index] != self.account.iban and self._get_amount(row):
```

Pylint (4.0.5)

[convention] Line too long (126/100)

(C0301)